### DISCUSSION TITLE
Charts

### DISCUSSION THREAD

**Discussion by Zefau** [Zefau/ioBroker.jarvis#887] [Q&A]
_Originally posted by @AFPass in https://github.com/Zefau/ioBroker.jarvis/issues/865#issuecomment-900414763_
> Ich konnte heute testen und die Alpha ließ sich installieren und danach normal ausführen.

> Andere Frage, die Grafik bei den Charts schaut jetzt anders aus. Bleibt das so oder kommt der Style von Version 2 wieder?

**Comment by Zefau** [comment]
Das Chart Modul ist von [Chart.js](https://www.chartjs.org/) auf [Echarts](https://echarts.apache.org/en/index.html) umgestellt, insofern sieht es anders aus.

Was genau meinst du mit Grafik / Style? Dann kann ich die Antwort ggf. präzisieren.

**Comment by AFPass** [comment]
In Version 2 sind die kurven "geschwungen" und nicht zu "zackig". Auch war die Linie etwas breiter.
Ich bin erstmal wieder auf Version 2. Ggf. mach ich noch mal Screenshots.

**Comment by Zefau** [comment]
in v3 gibt es die Möglichkeit zur Aggregation, womit Werte zusammengefasst werden. Dadurch sollte die Linie weicher / geschwungener werden. Die Dicke der Linie kann ich nochmal anpassen.

**Comment by AFPass** [comment]
Ok. Dann schau ich noch mal.
Danke.

**Comment by AFPass** [comment]
Warum wechselt Du von Chart.js auf ECharts? 
Irgendwie kann ich mich immer noch nicht damit anfreunden. 😬 Aber das ist jammern auf hohem Niveau. 👍

**Comment by Zefau** [comment]
Was ist der Vorteil von Chart.js zu Echarts?

**Reply by AFPass** [reply]
> Was ist der Vorteil von Chart.js zu Echarts?

🤦‍♂️ Du hast mich gefragt. 😬

Keine Ahnung, deswegen wollte ich wissen warum du wechselst.

**Comment by AFPass** [comment]
Ja

**Reply by Zefau** [reply]
?

**Comment by MarkSau** [comment]
Hi...Was muss ich den bei den neuen Echarts eintragen, damit die Y Achse sich dynamisch oder ab einem min. Wert anpasst. Beispiel: Ich habe einen Chart, der den Luftdruck anzeigt. Da dieser nie Null sein kann (dann sind wird alle tot), möchte ich entweder ein minimum angeben können oder es soll es soll sich automatisch anhand der Werte anpassen.  Ich werde aus der Echart Doku nicht wirklich schlau.

**Reply by Bernd9000** [reply]
> 
> 
> Hi...Was muss ich den bei den neuen Echarts eintragen, damit die Y Achse sich dynamisch oder ab einem min. Wert anpasst. Beispiel: Ich habe einen Chart, der den Luftdruck anzeigt. Da dieser nie Null sein kann (dann sind wird alle tot), möchte ich entweder ein minimum angeben können oder es soll es soll sich automatisch anhand der Werte anpassen. Ich werde aus der Echart Doku nicht wirklich schlau.

[#930](https://github.com/Zefau/ioBroker.jarvis/discussions/930)

**Comment by MarkSau** [comment]
Ich bin scheinbar zu doof die Optionen richtig zu setzten.  Ein Versuch, der nicht tut, aber vielleicht widerspiegelt was ich machen will.
{"setOption({yAxis.min}})": 800}
Ich verstehe einfach nicht, wo ich wann die verschiedenen Klammeren usw. setzten muss.  OK, man muss vielleicht JSON können, aber da haperts bei mir leider.

**Reply by mcuiobroker** [reply]
Die Einstellungen gehen doch nicht beim Chart, sondern nur bei HistoryGraph. Oder?

**Reply by MarkSau** [reply]
ja stimmt, bin ich dann hier falsch abgebogen?

**Reply by mcuiobroker** [reply]
Dann hier ein Beispiel:

`[ {    "position": "right",    
        "name": "Inzidenz",    
        "min": 0,    
        "max": 150,    
        "nameTextStyle": {
              "color": "red",      
              "backgroundColor": "grey",      
              "fontSize": 14    
        },    
        "axisLabel": {      
              "formatter": "{value} ml"    
        }  
  }]`

Oder für Dich: 
`[{"min":800}]`  nur mit/seit `v3.0.0-alpha.84` vorher nur `{"min":800}`

**Comment by mcuiobroker** [comment]
@Zefau Welchen Inhalt muss der DP für Chart haben?
![image](https://user-images.githubusercontent.com/84818934/135088930-d121b2c4-894b-410b-ab15-f6cb512857b6.png)
![image](https://user-images.githubusercontent.com/84818934/135089829-09b4ebd7-20d2-453b-be4c-7c3d46630441.png)

[12.5,13,15,17]? Funktioniert nicht. Muss es ein JSON sein?

**Reply by mcuiobroker** [reply]
Hat jemand ein Beispiel für mehrere Werte im Chart. Nicht HistoryGraph!
Hab mehrere Varianten versucht:
`[
  {
    "name": "Test",
    "value": 150
  },
  {
    "name": "Test2",
    "value": 120
  }
]` -> Keine Anzeige

`150,123,123,120` -> nur 150 wird angezeigt
Hab auch History Daten versucht -> keine Anzeige

**Comment by Bernd9000** [comment]
Für die Automatik:
![1 - Bild](https://user-images.githubusercontent.com/77005685/135102426-daedfe25-2975-4d95-af98-bf07cf88f627.jpg)

**Reply by MarkSau** [reply]
Danke..fluppt
